macros/src/auto_args.rs

```diff
@@ -76,21 +109,42 @@ macro_rules! build_rpc_trait {
 		$(#[$t_attr: meta])*
 		pub trait $name: ident {
 			type Metadata;
+
 			$(
-				$( #[doc=$m_doc:expr] )*
+				$( #[ doc=$m_doc:expr ] )*
 				#[ rpc( $($t:tt)* ) ]
 				fn $m_name: ident ( $($p: tt)* ) -> $result: tt <$out: ty, Error>;
 			)*
+
+			$(
+				#[ pubsub( $($pubsub_t:tt)+ ) ] {
+					$( #[ doc= $sub_doc:expr ] )*
+					#[ rpc( $($sub_t:tt)* ) ]
+					fn $sub_name: ident ( $($sub_p: tt)* );
+					$( #[ doc= $unsub_doc:expr ] )*
+					#[ rpc( $($unsub_t:tt)* ) ]
+					fn $unsub_name: ident ( $($unsub_p: tt)* ) -> $sub_result: tt <$sub_out: ty, Error>;
+				}
+			)*
+
 		}
 	) => {
 		$(#[$t_attr])*
 		pub trait $name: Sized + Send + Sync + 'static {
-			/// Requests metadata
-			type Metadata: ::jsonrpc_core::Metadata;
+			metadata! (
+				$( $sub_name )*
```

Comment:
a comment explaining that traits with subscriptions have different metadata than others might make sense :)